plugins/git4idea/src/git4idea/repo/GitRepositoryReader.java

```diff
@@ -442,14 +462,18 @@ else if (!resolved.containsKey(link)) {
     return resolved;
   }
 
+  private static String getRightRefName(RefInfo refInfo) {
+    return refInfo.referencedRef != null ? refInfo.referencedRef : refInfo.refName;
+  }
+
   @NotNull
-  private static Map<String, Hash> getResolvedHashes(@NotNull Map<String, String> data) {
-    Map<String, Hash> resolved = ContainerUtil.newHashMap();
+  private static Map<String, RefInfo> getResolvedHashes(@NotNull Map<String, String> data) {
+    Map<String, RefInfo> resolved = ContainerUtil.newHashMap();
     for (Map.Entry<String, String> entry : data.entrySet()) {
       String refName = entry.getKey();
       Hash hash = parseHash(entry.getValue());
```

Comment:
This is not a problem of this PR (since the problem already exists), mentioning here just for reference (mostly for myself): this method is potentially buggy, since it will treat a branch name matching `[a-zA-Z0-9]` as a hash.